lib/compress/index.js

```diff
@@ -1598,6 +1599,52 @@ def_optimize(AST_Switch, function(self, compressor) {
     while (i < len) eliminate_branch(self.body[i++], body[body.length - 1]);
     self.body = body;
 
+    // group equivalent branches so they will be located next to each other,
+    // that way the next micro-optimization will merge them.
+    // ** bail micro-optimization if not a simple switch case with breaks
+    if (body.every(branch =>
+        (
+            branch instanceof AST_Default || // default
+            !branch.expression.has_side_effects(compressor) // or no_side_effects
+        )
+            && (
+                branch.body.length === 0 // empty
+                || is_break(branch.body[branch.body.length - 1], compressor) // or has a break
+                || is_return(branch.body[branch.body.length - 1]) // or has a return
```

Comment:
```suggestion
                || branch.body.some(stmt => is_break(stmt, compressor)) // or has a break
                || branch.body.some(is_return) // or has a return
```

The break or return statements aren't always at the end

Reply:
Wouldn't this be better as just an `aborts` check?

Reply:
Oh, makes a lot of sense!

Reply:
(edited the suggestion)